### ISSUE TITLE
No attribute `node['to_str']' exists on the current node.

### ISSUE THREAD

**Issue by martinb3** [sethvargo/chef-sugar#85]
#75 / #76 broke some very simple use cases for me with chef-sugar 2.5.

For example, I had to make this change:
```
- Chef::Log.debug("redis_discovery found nodes #{redis_nodes.join(', ')}")
+ Chef::Log.debug("redis_discovery found nodes #{redis_nodes.map(&:name).join(', ')}")
```

To get rid of this exception:
```
 Failure/Error: ChefSpec::ServerRunner.new(platform: platform, version: version) do |node, server|
     Chef::Node::AttributeDoesNotExistError:
       No attribute `node['to_str']' exists on
       the current node. Please make sure you have spelled everything correctly.
```

Perhaps this is also a difference between Ruby 2.1.5 and 1.9.3-x branches, but it seems like the exception is misleading at best. I'm not sure what a better solution is, but I thought I'd bring it to your attention.

**Comment by sethvargo** [comment]
@martinb3 I don't even see where you are using `#deep_fetch!` in there... that is just mapping over node objects and relying on their underlying `#to_s` implementation to home it does the right thing. Your new solution is the proper thing to do. The `#to_s` implementation might have changed in Ruby 2.1.5 or it's not being implicitly called.

If this continues to be an issue once an official version of Chef that supports 2.1.5 is released, we can revisit it.

**Comment by martinb3** [comment]
@sethvargo OK, thank you for taking a look. I was under the impression `0.0.1` wasn't even a published version of chefspec (the tag does not exist on github). Also just to clarify, I consulted these RFC sections first:

> At the time of writing, Ruby 2.1 is the latest Ruby version and 2.0 is the "N-1" version.

https://github.com/opscode/chef-rfc/blob/master/rfc034-ruby-193-eol.md

> Chef Client omnibus packages ship with the Latest version of Ruby at the time of a major version bump. Latest & Latest - 1 versions of Ruby are supported. Ruby version bumps only happen at the time of Chef major version bumps.

https://github.com/opscode/chef-rfc/blob/master/rfc015-chef-12.md

I understood those to imply that chef 12 was released while ruby 2.1.5 was already out, so the official version of Chef supports Ruby 2.1 (with point releases).

FWIW, I do expect you'll see more of these. My colleagues and I discovered we had all been pinning chefspec forward without each other knowing it. 

Thanks for your time, again; it's very much appreciated!
